### ISSUE TITLE
[Dup Glitch] Player progress resetting.

### ISSUE THREAD

**Issue by neviskcubo** [GC-spigot/battle-pass#438]
### Describe the bug

Specific players are complaining that their progress is being reset when the server restarts.

### How to reproduce

The pass needs to be active in multi-server mode and one of the servers restarts, and for some reason some players' progress will be reset.

### Server Log

https://mcpaste.io/9b4a3ca4cfb3a978

**Comment by thiagogebrimm** [comment]
same problem, it's really annoying

**Comment by thiagogebrimm** [comment]
It seems that the bungecoord config doesn't work as well as it should

**Comment by neviskcubo** [comment]
the error remains, generating an absurd duplication of items as progress resets and players can get the rewards again, I don't understand how this error continues to be ignored...

**Comment by Timoti11** [comment]
me too...

**Comment by KPGTB** [comment]
Hey! I can't recreate that problem. Can you send video of that problem, or steps that need to be done to recreate that?

**Comment by Timoti11** [comment]
> Hey! I can't recreate that problem. Can you send video of that problem, or steps that need to be done to recreate that?

I can't make video with it, because it's randomly event

Steps:
1. Create same daily and weekly quests for each back-server
2. Link it with help of MySQL (MariaDB in my case)
3. Randomly after relogin check that some players have full reset of progress and levels

Paper, 1.20.2, offline-mode

**Comment by GC-spigot** [comment]
We're looking into this, sorry about the delay and thanks for the report. We'll keep in touch if we have follow up questions or when it's resolved.

**Comment by KPGTB** [comment]
Hey! How do you restart your server? I.e. by using restart/stop command, closing server using ^C in terminal or by killing process using i.e. commands to kill process, spamming ^C or closing terminal window?

**Comment by neviskcubo** [comment]
> Hey! How do you restart your server? I.e. by using restart/stop command, closing server using ^C in terminal or by killing process using i.e. commands to kill process, spamming ^C or closing terminal window?


Currently to manage my servers I use pterodactyl and for automatic restart I use: Ultimate auto restart, I don't know how it manages this restart, but I imagine it is the same effect as pressing "restart" on the ptero.

**Comment by GC-spigot** [comment]
@neviskcubo @Timoti11 @thiagogebrimm I'm testing this as well, I can't recreate it either just yet - but we won't give up until this is figured out.

I need few more questions answered?
- Does this reset all the rewards they have opened but still keep the points, or is all progress reset?
- When server restarts, are players automatically moved to the other server or are kicked from network?
- Does this happen just relogging without server restarts?

**Comment by neviskcubo** [comment]
> @neviskcubo @Timoti11 @thiagogebrimm I'm testing this as well, I can't recreate it either just yet - but we won't give up until this is figured out.
> 
> I need few more questions answered:
> 
> * Does this reset all the rewards they have opened but still keep the points, or is all progress reset? Completely reset
> * When server restarts, are players automatically moved to the other server or are kicked from network? Are moved to another server
> * Does this happen just relogging without server restarts? I didn't get any information about just relogging
> * How often does this happen? For a while it stopped happening, after updates to mysql, but recently it started happening again, with an average of 150 players online, we have 1~3 reset cases per day
> 
> Thank you so much guys, sorry about this problem and let's figure this one out together.

**Comment by neviskcubo** [comment]
The error is getting worse, I was using 4.4.2 I decided to update to 4.4.9, it reset the progress of almost all players on the server, I update their progress, a few hours pass, it resets again

From 2 days ago to today:
![image](https://github.com/GC-spigot/battle-pass/assets/86555997/9cf42da8-c86a-4e27-8ab6-b4d8130c9b58)

**Comment by KPGTB** [comment]
I'm sorry, but we still are trying to fix it, but we can't recreate that problem 😢

**Comment by neviskcubo** [comment]
> I'm sorry, but we still are trying to fix it, but we can't recreate that problem 😢

error persists > 2023 ... loading 2025

**Comment by KPGTB** [comment]
Hey! Can you update plugin to the latest version and check if there are any errors / warnings when affected players join the server? As I said, we are trying to fix this problem, but we can't reproduce it on our servers 😟

**Comment by neviskcubo** [comment]
I'm on the latest version and following the console, there are no errors recorded, do I need to activate an option to display the debug?

**Comment by KPGTB** [comment]
No, it should show an information when something go wrong with loading data

**Comment by neviskcubo** [comment]
> No, it should show an information when something go wrong with loading data

No error reported, I search for the key words: debug, sql, pass or player's own nick and no error returns

**Comment by thiagogebrimm** [comment]
> > I'm sorry, but we still are trying to fix it, but we can't recreate that problem 😢
> 
> error persists > 2023 ... loading 2025

This is sad, the plugin is only good in single server mode, when you activate multiserver this problem starts...

**Comment by thiagogebrimm** [comment]
'-' this problem not has solved

**Comment by GC-spigot** [comment]
@thiagogebrimm naybe with a next review on Spigot we'll reveal the fix. We cannot recreate it, as simple as that.

**Comment by thiagogebrimm** [comment]
Sad to know that you are not trying to improve the plugin, other people have had the same error for years, it makes me sad to see your ignorance :(

**Comment by neviskcubo** [comment]
Why is my issue closed if the error persists?

**Comment by KPGTB** [comment]
Hey! I'm sorry, but nobody from our development team can reproduce your error. We've tried a lot of times to reproduce & fix it, but we couldn't find the problem.

**Comment by neviskcubo** [comment]
> Hey! I'm sorry, but nobody from our development team can reproduce your error. We've tried a lot of times to reproduce & fix it, but we couldn't find the problem.

I understand, so the plugin has an active dup and no correction because they couldn't reproduce it? It's not just me who faces this

**Comment by KPGTB** [comment]
Hey! We've decided to implement new storage system into BattlePass which will improve data related stuff. It'll also definitively fix this glitch, so I reopened this issue. Currently, we are testing this system, so you can expect this feature in the next weeks

**Comment by KPGTB** [comment]
Hey! Should be fixed in 5.0.0 when using SQLite or MySQL-Improved storage method
